source/graphql/graphql.d

```diff
@@ -143,77 +143,64 @@ class GraphQLD(T, QContext = DefaultContext) {
 		}
 	}
 
-	Json getDefaultArgumentImpl(string typename, Type)(string type,
-			string field)
+	static Json getDefaultArgumentImpl(Type)(string field)
 	{
 		static if(isAggregateType!Type) {
-			if(typename == type) {
-				switch(field) {
-					static foreach(mem; __traits(allMembers, Type)) {
-						static if(isCallable!(
-								__traits(getMember, Type, mem))
-							)
-						{
-							case mem: {
-								alias parNames = ParameterIdentifierTuple!(
-										__traits(getMember, Type, mem)
-									);
-								alias parDef = ParameterDefaultValueTuple!(
-										__traits(getMember, Type, mem)
-									);
-
-								Json ret = Json.emptyObject();
-								static foreach(i; 0 .. parNames.length) {
-									static if(!is(parDef[i] == void)) {
-										ret[parNames[i]] =
-											serializeToJson(parDef[i]);
-									}
+			switch(field) {
+				static foreach(mem; __traits(allMembers, Type)) {
+					static if(isCallable!(
+							__traits(getMember, Type, mem))
+						)
+					{
+						case mem: {
+							alias parNames = ParameterIdentifierTuple!(
+									__traits(getMember, Type, mem)
+								);
+							alias parDef = ParameterDefaultValueTuple!(
+									__traits(getMember, Type, mem)
+								);
+
+							Json ret = Json.emptyObject();
+							static foreach(i; 0 .. parNames.length) {
+								static if(!is(parDef[i] == void)) {
+									ret[parNames[i]] =
+										serializeToJson(parDef[i]);
 								}
-								return ret;
 							}
+							return ret;
 						}
 					}
-					default: break;
 				}
+				default: break;
 			}
 		}
 		return Json.init;
 	}
 
 	Json getDefaultArguments(string type, string field) {
-		import graphql.traits : collectTypes;
-		switch(type) {
-			static foreach(Type; collectTypes!(T)) {{
-				case Type.stringof: {
-					Json tmp = getDefaultArgumentImpl!(Type.stringof, Type)(
-							type, field
-						);
-					if(tmp.type != Json.Type.undefined
-							&& tmp.type != Json.Type.null_)
-					{
-						return tmp;
-					}
-				}
-			}}
-			default: {}
+		import graphql.traits : execForAllTypes;
+		alias defaultArgFn = Json function(string) @safe;
+		static defaultArgFn[string] items;
+		if(items is null)
+		{
+			static void setupItems(T)(ref defaultArgFn[string] items)
+			{
+				items[T.stringof] = &getDefaultArgumentImpl!T;
+			}
+			execForAllTypes!(T, setupItems)(items);
+			// add entry points
+			static foreach(entryPoint; FieldNameTuple!T) {
+				items[entryPoint] =
```

Comment:
`setupItems` works on a type, and uses the type name as the key for the AA. In this case, we are using the member name as the key. So it can't be done that way.